core/src/main/java/org/elasticsearch/client/transport/TransportClient.java

```diff
@@ -168,9 +172,17 @@ public TransportClient build() {
                 final TransportProxyClient proxy = new TransportProxyClient(settings, transportService, nodesService,
                     actionModule.getActions().values().stream().map(x -> x.getAction()).collect(Collectors.toList()));
 
+                List<LifecycleComponent> pluginLifecycles = pluginComponents.stream()
```

Comment:
Or, er, do we need to make it a list at all? Maybe just add them all to the resourcesToClose?